.editorconfig

```diff
@@ -210,3 +210,214 @@ csharp_prefer_braces = false:none
+dotnet_diagnostic.RS1022.severity = none
+dotnet_diagnostic.RS1023.severity = none
+
+# Microsoft.Composition.Analyzers
+dotnet_diagnostic.RS0006.severity = error
+dotnet_diagnostic.RS0023.severity = error
+
+# Roslyn.Core
+dotnet_diagnostic.AD0001.severity = error
+
+# Roslyn.Diagnostic.Analyzers
+dotnet_diagnostic.RS0001.severity = warning
+dotnet_diagnostic.RS0002.severity = warning
+dotnet_diagnostic.RS0005.severity = warning
+dotnet_diagnostic.RS0016.severity = error
+dotnet_diagnostic.RS0017.severity = error
+dotnet_diagnostic.RS0022.severity = error
+dotnet_diagnostic.RS0024.severity = error
+dotnet_diagnostic.RS0025.severity = error
+dotnet_diagnostic.RS0026.severity = error
+dotnet_diagnostic.RS0027.severity = error
+dotnet_diagnostic.RS0030.severity = error
+dotnet_diagnostic.RS0031.severity = error
+dotnet_diagnostic.RS0033.severity = none        # Importing constructor should be [Obsolete]
+dotnet_diagnostic.RS0034.severity = none        # Style rule that enforces public MEF constructor marked with [ImportingConstructor]
+
+# System.Collections.Immutable.Analyzers
+dotnet_diagnostic.RS0012.severity = warning
+
+# System.Runtime.Analyzers
+dotnet_diagnostic.CA1305.severity = none
+dotnet_diagnostic.CA1307.severity = none
+dotnet_diagnostic.CA1308.severity = none
+dotnet_diagnostic.CA1810.severity = none
+dotnet_diagnostic.CA1816.severity = none
+dotnet_diagnostic.CA1825.severity = warning
+dotnet_diagnostic.CA2002.severity = none
+dotnet_diagnostic.CA2207.severity = none
+dotnet_diagnostic.CA2208.severity = none
+dotnet_diagnostic.CA2216.severity = none
+dotnet_diagnostic.CA2219.severity = none
+dotnet_diagnostic.CA2241.severity = none
+dotnet_diagnostic.CA2242.severity = none
+dotnet_diagnostic.RS0014.severity = warning
+
+# System.Runtime.InteropServices.Analyzers
+dotnet_diagnostic.CA1401.severity = none
+dotnet_diagnostic.CA2101.severity = none
+dotnet_diagnostic.RS0015.severity = warning
+
+# System.Threading.Tasks.Analyzers
+dotnet_diagnostic.RS0018.severity = warning
+
+# XmlDocumentationComments.Analyzers
+dotnet_diagnostic.RS0010.severity = warning
+
+# Microsoft.CodeAnalysis.CSharp.Features
+                                                # Name:                                         Before:                                             After:
+dotnet_diagnostic.IDE0001.severity = error      # Simplify names                                System.Version version;                             Version version;
+dotnet_diagnostic.IDE0002.severity = error      # Simplify (member access)                      System.Version.Equals("1", "2");                    Version.Equals("1", "2");
+dotnet_diagnostic.IDE0005.severity = error      # Using directive is unnecessary                using System.Text;
+dotnet_diagnostic.IDE0030.severity = error      # Use coalesce expression (nullable)            int? y = x.HasValue ? x.Value : 0                   int? y = x ?? 0;
+dotnet_diagnostic.IDE0030WithoutSuggestion.severity = error
```

Comment:
❔ Can you explain the need for this one?

Reply:
Nope. This was a largely mechanical translation from the .ruleset file. I'm not trying to reason about _why_ we have the settings we do. 😄

Reply:
I think we can follow up with another PR if we want to remove them.